### PR TITLE
🚧 Accept predefined email from URL

### PR DESCRIPTION
**Pull request by WeiJun0827**
截圖 2023-04-21 下午6 23 14
截圖 2023-04-21 下午6 23 32

Further UI improvement is welcome.
Not sure whether we should also provide some warning sentence, in case of abuse of this mode.

### CODE REVIEW

File: src/pages/settings/email/index.vue [hunk 1]
```diff
@@ -77,10 +80,18 @@ export default {
...
+    isClaimingPendingNftMode() {
+      return !!this.$route.query.claim_pending_nft && !this.hasEmail;
+    },
     isDisabledChangingEmail() {
       return !this.newEmail || this.newEmail === this.walletEmailUnverified;
     },
   },
+  mounted() {
+    if (this.isClaimingPendingNftMode) {
+      this.newEmail = this.$route.query.email;
```
Comment (nwingt) [inline src/pages/settings/email/index.vue:92]: it is more direct to inject it in `data()`

File: src/pages/settings/email/index.vue [hunk 2]
```diff
@@ -87,10 +87,15 @@ export default {
...
   },
-  mounted() {
-    if (this.isClaimingPendingNftMode) {
-      this.newEmail = this.$route.query.email;
-    }
+  watch: {
+    isClaimingPendingNftMode: {
+      immediate: true,
+      handler(newVal) {
+        if (newVal) {
```
Comment (nwingt) [inline src/pages/settings/email/index.vue:94]: Should check `this.$route.query.email` is not empty too?

**Review by nwingt — APPROVED** [review]

File: src/locales/en.json [hunk 3]
```diff
@@ -475,6 +475,7 @@
   "settings_email": "Email",
   "settings_email_change_email": "Change email",
   "settings_email_changing_email_submitted": "An verification email has been sent, please check your inbox.",
+  "settings_email_claim_pending_nft": "We will verify wallet with the email you provided on the previous credit card payment page.",
```
Comment (williamchong) [inline src/locales/en.json:478]:
```suggestion
  "settings_email_claim_pending_nft": "The email you provided during credit card payment will be used for verification.",

```

File: src/pages/settings/email/index.vue [hunk 4]
```diff
@@ -77,10 +80,23 @@ export default {
...
     isDisabledChangingEmail() {
       return !this.newEmail || this.newEmail === this.walletEmailUnverified;
     },
   },
+  watch: {
+    isClaimingPendingNftMode: {
+      immediate: true,
+      handler(newVal) {
+        if (newVal && this.$route.query.email) {
+          this.newEmail = this.$route.query.email;
```
Comment (williamchong) [inline src/pages/settings/email/index.vue:95]: `this.newEmail = this.$route.query.email;` we can always assign this in data regard less `isClaimingPendingNftMode` or not

**Review by williamchong — CHANGES_REQUESTED** [review]

**Review by williamchong — APPROVED** [review]

**Pull request merged**